include/ginkgo/core/base/array.hpp

```diff
@@ -326,14 +352,14 @@ class Array {
             this->resize_and_reset(0);
             return *this;
         }
-        if (exec_ == other.get_executor() &&
-            data_.get_deleter().target_type() != typeid(view_deleter)) {
-            // same device and not a view, only move the pointer
+        if (exec_ == other.get_executor()) {
+            // same device, only move the pointer
             using std::swap;
             swap(data_, other.data_);
             swap(num_elems_, other.num_elems_);
+            other.resize_and_reset(0);
```

Comment:
nit: could you actually use the `clear()` function here? For some reason, we use `resize_and_reset(0)` in a lot of places instead of `clear()`, even though `clear()` has less side effects, is more light weight and more descriptive.